src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,68 @@
+package duke.task;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected TaskType taskType;
+
+    public Task() {}
+
+    public Task(String description, TaskType taskType) {
+        this.description = description;
+        this.isDone = false;
+        this.taskType = taskType;
+    }
+
+    public void setAsDone() {
+        this.isDone = true;
+    }
+
+    public void setAsUndone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Returns symbol for printing task based on its type.
+     *
+     * @return [type] where type corresponds to task type.
+     */
+    public String getSymbol() {
```

Comment:
Would it be better if getSymbol is an abstract method implemented by the other Task sub-classes?